### ISSUE TITLE
Narrow down logging of configuration problems reported on requests

### ISSUE THREAD

**Issue by toniedzwiedz** [Cognifide/knotx#271]
I mistyped a piece of configuration (`knotx.io.knotx.KnotxServer` instead of `knotx:io.knotx.KnotxServer`) while following the [REST service tutorial](http://knotx.io/hello-rest-service/) and got the following errors:

```
2017-02-24 21:32:19 [vert.x-eventloop-thread-1] WARN  io.knotx.server.KnotxSplitterHandler - Requested path /html/books.html is not supported based on configuration
2017-02-24 21:32:19 [vert.x-eventloop-thread-1] WARN  io.knotx.server.KnotxSplitterHandler - Requested path /favicon.ico is not supported based on configuration
```

My logback configuration is as follows:

```
<configuration>

  <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
    <layout class="ch.qos.logback.classic.PatternLayout">
      <Pattern>
        %d{yyyy-MM-dd HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n
      </Pattern>
    </layout>
  </appender>

  <logger name="io.knotx" level="DEBUG"
          additivity="false">
    <appender-ref ref="STDOUT"/>
  </logger>

  <root level="DEBUG">
    <appender-ref ref="STDOUT"/>
  </root>

</configuration>
```

Is it possible to enable more verbose output and narrow down the configuration issue to a specific section? Maybe list all paths successfully registered together with their corresponding connectors?

**Comment by toniedzwiedz** [comment]
Upon closer inspection, this looks like an addition that could be made in `SupportedMethodsAndPathsHandler#handle(RoutingContext context)`

I'll think of a clean way to log the paths actually enabled for `KnotxServer` on a detailed log level.

**Comment by toniedzwiedz** [comment]
I added some extra logging inside `SupportedMethodsAndPathsHandler` and I noticed a couple of things:

1. If no valid configuration is provided for `io.knotx.KnotxServer`, Knot.x falls back to the following configuration:
   ```
    {
      "GET" : [ {
        "path" : "/content/.*",
        "address" : "knotx.knot.service",
        "onTransition" : {
          "next" : {
            "path" : null,
            "address" : "knotx.knot.handlebars",
            "onTransition" : { }
          }
        }
      } ]
    }
   ```

   The way I implemented the extra debug information is by printing the whole config from `"knotx:io.knotx.KnotxServer".options.config.routing`. For now, I used some dirty code just to see what's going on. 

   If the default is applied consistently, a single line warning that Knot.x falls back to the default configuration would be enough to show that the configuration is missing or malformed. However, I'm having a hard time finding where the default is applied in the core Knot.x code.

 2. The logger defined in `SupportedMethodsAndPathsHandler` seems to be using the wrong class.
     `private static final Logger LOGGER = LoggerFactory.getLogger(KnotxSplitterHandler.class);`

 3. The following code is evaluated even if we already know the request should fail based on the value of `shouldRejectMethod`. Not sure if this is a problem but a minor refactor might potentially improve performance somewhat, depending on the number of paths configured.

        boolean shouldRejectPath = engineRoutingValues.stream().noneMatch(
            routingEntries -> routingEntries.stream()
                .anyMatch(item -> context.request().path().matches(item.path()))
        );
